docs/rules/no-if.md

```diff
@@ -1,6 +1,10 @@
 # Disallow conditional logic (`no-if`)
 
-## Deprecated
+<!-- RULE_NOTICE -- Generated by `yarn tools:regenerate-docs` -->
+
+❌ This rule is deprecated.
```

Comment:
nit: doesn't have to be done in this PR, but a good followup could be adding "in favor of ..." if there's a suggested replacement

Reply:
Definitely like this idea so I went ahead and updated the notice to link to the replacement rule(s).